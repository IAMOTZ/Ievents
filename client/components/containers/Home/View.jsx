import React from 'react';
import { Link } from 'react-router-dom';
import { RegularTopNavigation } from '../../common/TopNavigation';

const Home = () => (
  <div id="home-container">
    <RegularTopNavigation />
    <div className="dark-overlay">
      <div
        id="page-content"
        className="d-flex flex-column align-items-center text-center"
      >
        <h1 className="display-2 text-center">Welcome to<strong> iEvents</strong>
        </h1>
        <p className="display-4 d-none d-sm-block">Get the perfect center for your events</p>
        <p
          className="display-4 d-block d-sm-none"
          style={{ fontSize: '2.0rem' }}
        >Get the perfect center for your events
        </p>
        <Link to="/explore/centers" className="btn btn-primary mt-3">
          <i className="fa fa-bank fa-fw" aria-hidden="true" />&nbsp; Checkout Our Centers
        </Link>
      </div>
    </div>
  </div>
);

export default Home;
