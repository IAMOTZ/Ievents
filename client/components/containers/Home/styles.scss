$custom-black: rgba(0,0,0, 0.9);
$custom-blue: rgb(0, 142, 214);

#home-container {
  background: url('/images/eventCenter1.jpeg') no-repeat;
  min-height: 100vh;
  background-size: cover;
  background-attachment: fixed;
  border-bottom: 2px solid $custom-blue;
  .dark-overlay {
    background-color: rgba(0,0,0,0.5);
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    min-height: 100vh;
  }
  #page-content {
    padding-top: 100px;
    color: white;
  }
}
