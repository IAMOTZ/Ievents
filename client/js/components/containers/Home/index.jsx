import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import View from './View';

@connect(store => (
  {
    isUserAuthenticated: store.authReducer.loggingUserResolved,
  }
))
class Home extends React.Component {
  render() {
    if (this.props.isUserAuthenticated) {
      return <Redirect to="/addEvent" />;
    }
    return <View />;
  }
}

Home.defaultProps = {
  isUserAuthenticated: false,
};

Home.propTypes = {
  isUserAuthenticated: PropTypes.bool,
};

export default Home;
