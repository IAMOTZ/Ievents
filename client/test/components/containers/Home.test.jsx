import React from 'react';
import Home from '../../../components/containers/Home';
import HomeView from '../../../components/containers/Home/View';

describe('<Home />', () => {
  const props = {
    isUserAuthenticated: false,
  };
  const alterProps = newProps => ({ ...props, ...newProps });
  describe('Rendering:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Home {...props} />);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should redirect to centers page if user is authenticated', () => {
      wrapper.setProps(alterProps({ isUserAuthenticated: true }));
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('<HomeView />', () => {
  describe('Rendering', () => {
    it('should render correctly', () => {
      const wrapper = shallow(<HomeView />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
