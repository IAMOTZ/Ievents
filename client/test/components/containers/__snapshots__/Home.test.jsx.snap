// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Rendering: should redirect to centers page if user is authenticated 1`] = `
<Redirect
  push={false}
  to="/centers"
/>
`;

exports[`<Home /> Rendering: should render correctly 1`] = `<Home />`;

exports[`<HomeView /> Rendering should render correctly 1`] = `
<div
  id="home-container"
>
  <RegularTopNavigation />
  <div
    className="dark-overlay"
  >
    <div
      className="d-flex flex-column align-items-center text-center"
      id="page-content"
    >
      <h1
        className="display-2 text-center"
      >
        Welcome to
        <strong>
           iEvents
        </strong>
      </h1>
      <p
        className="display-4 d-none d-sm-block"
      >
        Get the perfect center for your events
      </p>
      <p
        className="display-4 d-block d-sm-none"
        style={
          Object {
            "fontSize": "2.0rem",
          }
        }
      >
        Get the perfect center for your events
      </p>
      <Link
        className="btn btn-primary mt-3"
        replace={false}
        to="/explore/centers"
      >
        <i
          aria-hidden="true"
          className="fa fa-bank fa-fw"
        />
        Â  Checkout Our Centers
      </Link>
    </div>
  </div>
</div>
`;
